lib/hammer_cli_foreman_discovery/discovery_rule.rb

```diff
@@ -0,0 +1,77 @@
+require 'hammer_cli'
+require 'hammer_cli_foreman'
+require 'hammer_cli_foreman_discovery/discovery_references'
+
+module HammerCLIForemanDiscovery
+
+  module CommonDiscoveryRuleUpdateOptions
+
+    def self.included(base)
+      base.option "--hosts_limit", "HOSTS_LIMIT", " "
```

Comment:
Btw I think you can define `:attribute_name => :option_max_count` and avoid the `request_params` override.